pkg/metrics/metrics.go

```diff
@@ -251,6 +252,60 @@ func (c *Config) validate(ctx context.Context) (bool, error) {
 	return true, nil
 }
 
+// RemoveStaleResources deletes stale resources that could be left behind during the
+// upgrade process due to the renaming of resources between current and previous operator version
+func (c *Config) RemoveStaleResources(ctx context.Context) {
+	wmcoNamespace, err := c.CoreV1().Namespaces().Get(ctx, c.namespace, metav1.GetOptions{})
+	if err != nil {
+		log.Error(err, "error getting operator namespace")
+	}
+
+	// staleResourceName is the metrics object name created for Prometheus monitoring by previous operator versions
+	staleResourceName := wmcoNamespace.GetName() + "-metrics"
+
+	// remove stale service objects
+	if _, err := c.CoreV1().Services(c.namespace).Get(ctx, staleResourceName, metav1.GetOptions{}); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
err is not being created again here which causes issue when using err nil check for deletion, adding a fix for this.

Reply:
Won't `err` always get overwritten in the subsequent calls?

Reply:
NVM, I see the problem.